uint32 time2p5ns
uint32 time1ns
uint8 flag
